atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueService.java

```diff
@@ -48,6 +51,12 @@
 
 @SuppressFBWarnings("SLF4J_ILLEGAL_PASSED_CLASS")
 public abstract class AbstractKeyValueService implements KeyValueService {
+
+    private static final LoadingCache<TableReference, String> internalTableNames = Caffeine.newBuilder()
+            .weigher((TableReference key, String value) -> Ints.saturatedCast(key.sizeInBytes() + value.length()))
+            .maximumWeight(1_000_000)
+            .build(AbstractKeyValueService::computeInternalTableName);
```

Comment:
Related to #6397 wondering if we should push the internal table name as a lazily computed field inside TableReference itself, and just intern the TableReference instances